### PR TITLE
feat: Add Offer Settings modal

### PR DESCRIPTION
**Pull request by MrSltun**
This PR resolves [EMI-1708], [EMI-1716], [EMI-1776] 

> [!NOTE]
> This feature is hidden behind `AREnableArtworkListOfferability` feature flag

### Description
This PR adds a new feature to update the user's save lists to be shared or not shared with partners
It also adds the ~~Lock~~ EyeClosed icon in the artwork lists to indicate shareability with partners

#### Demo (v1.1)

https://github.com/artsy/eigen/assets/20655703/8aa2c670-1063-46b5-a3d1-5bd629cec23f

### Screenshots (New)

| | Android | iOS |
|---|---|---|
| Saves screen | image | image |
| Offer Settings modal view | image | image |
| Add artwork to list modal view | image | image |
| Shared list screen | image | image |
| Unshared list screen | image | image |

### PR Checklist

- [X] I have tested my changes on **iOS** and **Android**.
- [X] I hid my changes behind a **[feature flag]**, or they don't need one.
- [X] I have included **screenshots** or **videos**, or I have not changed the UI.
- [ ] I have added **tests**, or my changes don't require any.
- [X] I added an **[app state migration]**, or my changes do not require one.
- [X] I have documented any **follow-up work** that this PR will require, or it does not require any.
- [X] I have added a **changelog entry** below, or my changes do not require one.

### To the reviewers 👀

- [ ] I would like **at least one** of the reviewers to **run** this PR on the simulator or device.

<details><summary>Changelog updates</summary>

#### Cross-platform user-facing changes

- Add Offer Settings feature in Saves - mrsltun

#### iOS user-facing changes

-

#### Android user-facing changes

-

#### Dev changes

-

</details>

Need help with something? Have a look at our [docs], or get in touch with us.

[app state migration]: ../blob/main/docs/adding_state_migrations.md
[feature flag]: ../blob/main/docs/developing_a_feature.md
[docs]: ../blob/main/docs/README.md

[EMI-1708]: https://artsyproduct.atlassian.net/browse/EMI-1708?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
[EMI-1716]: https://artsyproduct.atlassian.net/browse/EMI-1716?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

[EMI-1776]: https://artsyproduct.atlassian.net/browse/EMI-1776?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Comment by ArtsyOpenSource** [comment]
### This PR contains the following changes:

- Cross-platform user-facing changes (Add Edit List Privacy feature in Saves - mrsltun)
<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/js">dangerJS</a> against 05ecf7fe7d02073caa515b18a5e261a96aee5708
</p>

File: src/app/Components/ArtworkLists/ArtworkListsContext.tsx [hunk 1]
```diff
@@ -265,15 +267,70 @@ const reducer = (state: ArtworkListState, action: ArtworkListAction): ArtworkLis
         ...state,
         hasUnsavedChanges: action.payload,
       }
+    case "SET_OFFER_SETTINGS_VIEW_VISIBLE":
+      return {
+        ...state,
+        artworkListOfferSettingsViewVisible: action.payload,
+      }
+    case "SHARE_OR_KEEP_ARTWORK_LIST":
```
Comment (starsirius) [inline src/app/Components/ArtworkLists/ArtworkListsContext.tsx:275]: To be consistent, maybe "SHARE_OR_KEEP_ARTWORK_LIST_PRIVATE"?

File: src/app/Components/ArtworkLists/ArtworkListsContext.tsx [hunk 2]
```diff
@@ -24,12 +25,15 @@ export interface ArtworkListsProviderProps {
...
   createNewArtworkListViewVisible: false,
+  artworkListOfferSettingsViewVisible: false,
   artwork: null,
   artworkListID: null,
   recentlyAddedArtworkList: null,
   selectedTotalCount: 0,
   addingArtworkLists: [],
   removingArtworkLists: [],
+  keepingArtworkListsPrivate: [],
+  sharingArtworkLists: [],
```
Comment (starsirius) [inline src/app/Components/ArtworkLists/ArtworkListsContext.tsx:36]: #non-blocking: Using 2 arrays to maintain the state of list privacy for each list may require more work (e.g. needing to remove one item from one array and add it to the other) and less efficient. Potentially a single hash/object with `{ <artwork-list-id>: <list-privacy> }` could be easier to maintain.

But I see these follow the `adding/removingArtworkLists` pattern above, so it's fine to keep it consistent.

**Comment by starsirius** [comment]
Looking great! 💪

File: src/app/Components/ArtworkLists/views/SelectArtworkListsForArtworkView/components/ArtworkListItem.tsx
Comment (MrSltun) [inline src/app/Components/ArtworkLists/views/SelectArtworkListsForArtworkView/components/ArtworkListItem.tsx:1]: This file was not removed, it was only moved to a `src/app/Components/ArtworkLists/components/ArtworkListItem.tsx`

File: src/app/Components/LineGraph/LineGraphCategoryPicker.tsx [hunk 3]
```diff
@@ -56,9 +56,9 @@ export const LineGraphCategoryPicker: React.FC<LineGraphCategoryPickerProps> = m
         }
         let left = 20
         for (let i = 0; i < selectedIndex; i++) {
-          left += categoryLayouts[i]!.width
+          left += categoryLayouts[i]?.width ?? 0
         }
-        const center = left + categoryLayouts[selectedIndex]!.width / 2
+        const center = left + (categoryLayouts[selectedIndex]?.width ?? 0) / 2
```
Comment (MrSltun) [inline src/app/Components/LineGraph/LineGraphCategoryPicker.tsx:61]: Updated these to fix linting issues

File: src/app/Components/ArtworkLists/views/SelectArtworkListsForArtworkView/useSaveArtworkListsChanges.ts [hunk 4]
```diff
@@ -16,15 +16,15 @@ export const useSaveArtworkListsChanges = (options?: Options) => {
...
+  const artwork = state.artwork as ArtworkEntity
   const [commit, inProgress] = useUpdateArtworkListsForArtwork(artwork.id)
 
   const trackAddedArtworkToArtworkLists = () => {
     const event: AddedArtworkToArtworkList = {
       action: ActionType.addedArtworkToArtworkList,
       context_owner_id: analytics.contextScreenOwnerId,
       context_owner_slug: analytics.contextScreenOwnerSlug,
-      context_owner_type: analytics.contextScreenOwnerType!,
+      context_owner_type: analytics.contextScreenOwnerType || OwnerType.artwork,
```
Comment (MrSltun) [inline src/app/Components/ArtworkLists/views/SelectArtworkListsForArtworkView/useSaveArtworkListsChanges.ts:27]: Also fixing linting issues

File: src/app/Components/ArtworkLists/ArtworkListsContext.tsx [hunk 5]
```diff
@@ -19,17 +20,21 @@ export interface ArtworkListsProviderProps {
   // Needs for tests
   artwork?: ArtworkEntity
   selectArtworkListsViewVisible?: boolean
+  artworkListOfferSettingsViewVisible?: boolean
```
Comment (damassi) [inline src/app/Components/ArtworkLists/ArtworkListsContext.tsx:23]: This is certainly not for this PR or this team even, but wanted to drop by with a PSA that this whole file could be radically simplified by using easy peasy, which we already have in Eigen: https://github.com/artsy/volt-v2/blob/main/docs/best_practices.md#state

**Pull request merged**